ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/P2TargetUtils.java

```diff
@@ -1017,15 +1049,16 @@ static IQueryable<IInstallableUnit> getQueryableMetadata(URI[] repos, IProgressM
 		int repoCount = repos.length;
 		SubMonitor subMonitor = SubMonitor.convert(monitor, repoCount * 2);
 
-		Set<IRepositoryReference> seen = new HashSet<>();
```

Comment:
Maybe I have missed it, but what's the reason for the seen set to be re-created for each repository in a container?
If multiple repos referenced the same third repo it can now end up multiple times in the resulting list, can't it?

Reply:
I was simply trying to clarify that seen is not used in this function at all by inlining it. However, I didn't notice the once vs every iteration difference. I'll have to revert that one line as this changes the behavior

Reply:
See https://github.com/eclipse-pde/eclipse.pde/pull/1257